pkg/k8s/portforward.go

```diff
@@ -162,6 +163,46 @@ func (pf *PortForward) Run() error {
 	return fw.ForwardPorts()
 }
 
+// Init creates and runs a port-forward connection, waits for it to be ready, and returns
+// its URL and http client.
+func (pf *PortForward) Init(controlPlaneNamespace string, kubeAPI *KubernetesAPI) (*url.URL, *http.Client, error) {
```

Comment:
👍 I like moving this logic into `portforward.go`, even if we only end up calling this from one place.

But it looks like you're not using the `controlPlaneNamespace` param that's being passed in, and we're already passing `kubeAPI` when `NewPortForward` is called. If we persist that param on the `PortForward` struct, then you won't need to pass it here.

I also wonder if we could make this func more generic so that we could use it in `cli/cmd/metrics.go` and `cli/cmd/dashboard.go`.

Reply:
@klingerf I've created a separate PR for the port-forwarding refactoring: #2976
Looking forward for your comments :wink: